symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java

```diff
@@ -74,4 +74,11 @@ public MessageService messages(Obo.Handle oboHandle) throws AuthUnauthorizedExce
   public DatafeedService datafeed() {
     return this.datafeedService;
   }
+
+  protected OboAuthenticator getOboAuthenticator() throws AuthInitializationException {
+    if (this.oboAuthenticator == null) {
+      throw new AuthInitializationException("OBO is not configured", null);
+    }
+    return this.oboAuthenticator;
+  }
```

Comment:
I'd write something more Java8 :) 
```java
  protected OboAuthenticator getOboAuthenticator() {
    return Optional.ofNullable(this.oboAuthenticator)
        .orElseThrow(() -> new IllegalStateException("OBO is not configured."));
  }
```

As well, the `AuthInitializationException` is not really designed to handle this kind of error. Either we create a dedicated exception, or we keep the `IllegalStateException`. Anyway, that's the kind of error that a developer should see right when starting the application

Reply:
At the cost of allocating memory for the optional! :p

Reply:
Aren't you ex-C++ dev?!